src/fuzz/fuzz-calendarspec.c

```diff
@@ -0,0 +1,28 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "alloc-util.h"
+#include "calendarspec.h"
+#include "fd-util.h"
+#include "fuzz.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+        _cleanup_(calendar_spec_freep) CalendarSpec *cspec = NULL;
+        _cleanup_free_ char *str = NULL, *p = NULL;
+
+        if (!getenv("SYSTEMD_LOG_LEVEL"))
+                log_set_max_level(LOG_CRIT);
+
+        str = malloc0(sizeof(*data) * (size + 1));
+        assert_se(str != NULL);
+        memcpy_safe(str, data, size);
+        str[size] = '\0';
+
+        (void) calendar_spec_from_string(str, &cspec);
+        if (cspec) {
```

Comment:
Maybe it'd be better to use `if (calendar_spec_from_string() >= 0)` ? This tests that `cspec` is set on non-failure.

Reply:
Definitely makes sense.